zebra-state/src/service/finalized_state/disk_format/block.rs

```diff
@@ -15,21 +15,56 @@ use zebra_chain::{
     transaction,
 };
 
-use crate::service::finalized_state::disk_format::{FromDisk, IntoDisk, IntoDiskFixedLen};
+use crate::service::finalized_state::disk_format::{
+    expand_zero_be_bytes, truncate_zero_be_bytes, FromDisk, IntoDisk,
+};
 
 #[cfg(any(test, feature = "proptest-impl"))]
 use proptest_derive::Arbitrary;
 
+/// The maximum value of an on-disk serialized [`Height`].
+///
+/// This allows us to store [`OutputIndex`]es in 8 bytes,
+/// which makes database searches more efficient.
+///
+/// # Consensus
+///
+/// This maximum height supports on-disk storage of blocks until around 2050.
+///
+/// Since Zebra only stores fully verified blocks on disk, blocks with heights
+/// larger than this height are rejected before reaching the database.
+/// (It would take decades to generate a valid chain this high.)
+pub const MAX_ON_DISK_HEIGHT: Height = Height(1 << (HEIGHT_DISK_BYTES * 8));
```

Comment:
Should we not subtract a one here?

Reply:
Nice catch!

This also means the tests are wrong.

Reply:
Fixed in 691d1e710